bootstrapping/bootstrapping.go

```diff
@@ -0,0 +1,428 @@
+	revisionID string
+	filename   string
+	filePath   string
+	rawData    *json.RawMessage
+}
+
+type policyStatus struct {
+	id          string
+	revisionID  string
+	name        string
+	filename    string
+	tarballURL  string
+	queryURL    string
+	tarballPath string
+	folderPath  string
+
+	downloaded   bool
+	uncompressed bool
+	executed     bool
+	logged       bool
+}
+
+// Handle signals
+func handleSysSignals(cancelFunc context.CancelFunc) {
+	log.Debug("handleSysSignals")
+
+	gracefulStop := make(chan os.Signal, 1)
+	signal.Notify(gracefulStop, syscall.SIGTERM, syscall.SIGINT, syscall.SIGKILL)
+	log.Debug("Ending, signal detected:", <-gracefulStop)
+	cancelFunc()
+}
+
+// Returns the full path to the tmp folder joined with pid management file name
+func getProcessIDFilePath() string {
+	return strings.Join([]string{os.TempDir(), string(os.PathSeparator), ProcessIDFile}, "")
+}
+
+// Returns the full path to the tmp folder
+func getProcessingFolderFilePath() string {
+	dir := strings.Join([]string{os.TempDir(), string(os.PathSeparator), "imco", string(os.PathSeparator)}, "")
+	os.Mkdir(dir, 0777)
+	return dir
+}
+
+// Start the bootstrapping process
+func start(c *cli.Context) error {
+	log.Debug("start")
+
+	formatter := format.GetFormatter()
+	if err := utils.SetProcessIdToFile(getProcessIDFilePath()); err != nil {
+		formatter.PrintFatal("cannot create the pid file", err)
+	}
+
+	timingInterval := c.Int64("time")
+	if !(timingInterval > 0) {
+		timingInterval = DefaultTimingInterval
+	}
+	// Adds a random value to the given timing interval!
+	// Sleep for a configured amount of time plus a random amount of time (10 minutes plus 0 to 5 minutes, for instance)
+	timingInterval = timingInterval + int64(rand.New(rand.NewSource(time.Now().UnixNano())).Intn(DefaultRandomMaxThreshold)*60)
+	log.Debug("time interval:", timingInterval)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go handleSysSignals(cancel)
+
+	bootstrappingRoutine(ctx, c, timingInterval)
+
+	return nil
+}
+
+// Stop the bootstrapping process
+func stop(c *cli.Context) error {
+	log.Debug("cmdStop")
+
+	formatter := format.GetFormatter()
+	if err := utils.StopProcess(getProcessIDFilePath()); err != nil {
+		formatter.PrintFatal("cannot stop the bootstrapping process", err)
+	}
+
+	log.Info("Bootstrapping routine successfully stopped")
+	return nil
+}
+
+// Main bootstrapping background routine
+func bootstrappingRoutine(ctx context.Context, c *cli.Context, timingInterval int64) {
+	log.Debug("bootstrappingRoutine")
+
+	//formatter := format.GetFormatter()
+	bootstrappingSvc, formatter := cmd.WireUpBootstrapping(c)
+	commandProcessed := make(chan bool, 1)
+
+	// initialization
+	currentTicker := time.NewTicker(time.Duration(timingInterval) * time.Second)
+	for {
+		go processingCommandRoutine(bootstrappingSvc, formatter, commandProcessed)
+
+		log.Debug("Waiting...", currentTicker)
+
+		select {
+		//case <-commandProcessed:
```

Comment:
So far, not evaluated

Reply:
I don't think you will need this, you can process the command on the same goroutine and just wait for the 10 minutes or the context to be cancelled afterwards.